test/expressions.ml

```diff
@@ -1376,3 +1441,83 @@ let%expect_test "Lists and Arrays" =
     │ Station 10 ┆ 17    ┆ 13    ┆ 10    ┆ [0.33, 0.67, 1.0]  │
     └────────────┴───────┴───────┴───────┴────────────────────┘ |}]
 ;;
+
+let%expect_test "join_asof" =
+  let thirds = Series.float "a" [ 33.3; 66.6; 99.9 ] in
+  let tens = Series.float "b" [ 10.; 20.; 30.; 40.; 50.
+                              ; 60.; 70.; 80.; 90.; 100.] in
+  let tens = Data_frame.create_exn [tens] |> Data_frame.lazy_ in
+  let thirds = Data_frame.create_exn [thirds] |> Data_frame.lazy_ in
+  List.iter [`Backward; `Forward; `Nearest] ~f:(fun strategy ->
+      let joined = Lazy_frame.join'
+        thirds
+        ~other:tens
+        ~left_on: [Expr.col "a" |> Expr.set_sorted_flag ~sorted:Ascending]
+        ~right_on:[Expr.col "b" |> Expr.set_sorted_flag ~sorted:Ascending]
+        ~how:(As_of { strategy
+                    ; tolerance = None
+                    ; left_by   = None
+                    ; right_by  = None })
+      in
+      print_s ([%sexp_of : [`Backward | `Forward | `Nearest]] strategy);
+      Lazy_frame.collect_exn joined |> Data_frame.print);
+  [%expect {|
+    Backward
+    shape: (3, 2)
+    ┌──────┬──────┐
+    │ a    ┆ b    │
+    │ ---  ┆ ---  │
+    │ f64  ┆ f64  │
+    ╞══════╪══════╡
+    │ 33.3 ┆ 30.0 │
+    │ 66.6 ┆ 60.0 │
+    │ 99.9 ┆ 90.0 │
+    └──────┴──────┘
+    Forward
+    shape: (3, 2)
+    ┌──────┬───────┐
+    │ a    ┆ b     │
+    │ ---  ┆ ---   │
+    │ f64  ┆ f64   │
+    ╞══════╪═══════╡
+    │ 33.3 ┆ 40.0  │
+    │ 66.6 ┆ 70.0  │
+    │ 99.9 ┆ 100.0 │
+    └──────┴───────┘
+    Nearest
+    shape: (3, 2)
+    ┌──────┬───────┐
+    │ a    ┆ b     │
+    │ ---  ┆ ---   │
+    │ f64  ┆ f64   │
+    ╞══════╪═══════╡
+    │ 33.3 ┆ 30.0  │
+    │ 66.6 ┆ 70.0  │
+    │ 99.9 ┆ 100.0 │
+    └──────┴───────┘ |}]
+;;
+
+let%expect_test "take" =
+  let values = Series.float "val" [ 1.; 2.; 3.; 4.; 5.; 6. ] in
+  let idxs = Series.int "idx"   [ 1;  0;  1;  2;  0;  2 ] in
+  let ldf = Data_frame.create_exn[ values; idxs ] |> Data_frame.lazy_ in
+  let filtered = Lazy_frame.filter ldf ~predicate:(
+      (* Remove index 0, keep index 1 and 2 *)
+      Expr.series (Series.bool "filter" [ false; true; true ])
+      |> Expr.take ~idx:(Expr.col "idx"))
```

Comment:
Could we do something like `~predicate:Expr.( ... )` so we can reduce the number of mentions of `Expr`?

Reply:
Done